linker/jvm/src/main/scala/org/scalajs/linker/irio/FileVirtualIRFiles.scala

```diff
@@ -56,46 +57,46 @@ object FileScalaJSIRContainer {
 
 final class FileVirtualJarScalaJSIRContainer(file: File)
     extends FileVirtualFile(file) with ScalaJSIRContainer {
+  private[this] val zipFile = new SharedZipFile(file)
+
   def sjsirFiles: List[VirtualScalaJSIRFile] = {
-    val stream = new ZipInputStream(new BufferedInputStream(new FileInputStream(file)))
+    val pinnedVersion = version
+    val zf = zipFile.acquire()
     try {
-      val buf = new Array[Byte](4096)
-
-      @tailrec
-      def readAll(out: OutputStream): Unit = {
-        val read = stream.read(buf)
-        if (read != -1) {
-          out.write(buf, 0, read)
-          readAll(out)
-        }
-      }
+      zf.entries().asScala
+        .withFilter(_.getName.endsWith(".sjsir"))
+        .map(entryFile(_, pinnedVersion))
+        .toList
+    } finally {
+      zipFile.release()
+    }
+  }
 
-      def makeVF(e: ZipEntry) = {
-        val size = e.getSize
-        val out =
-          if (0 <= size && size <= Int.MaxValue) new ByteArrayOutputStream(size.toInt)
-          else new ByteArrayOutputStream()
+  private def entryFile(entry: ZipEntry, fileVersion: Option[String]) = {
+    val filePath = s"$path:${entry.getName()}"
 
+    def withInputStream[A](f: InputStream => A) = {
+      val zf = zipFile.acquire()
+      try {
+        val in = zf.getInputStream(entry)
```

Comment:
Is this actually safe to use in concurrent threads on the same `ZipFile`? If not, I believe the thread should keep the lock on the `SharedZipFile` while reading the input stream for an entry.

Reply:
Hmmm... Good question. The internet seems not to have any info about this (only about `ZipFileInputStream`).